src/olympia/editors/tests/test_models.py

```diff
@@ -1016,6 +1019,74 @@ def test_calculate_weight_uses_native_messaging(self):
         assert summary.weight == 20
         assert weight_info['uses_native_messaging'] == 20
 
+    def test_calculate_weight_uses_remote_scripts(self):
+        validation_data = {
+            'messages': [{
+                'id': ['REMOTE_SCRIPT'],
+            }]
+        }
+        self.file_validation.update(validation=json.dumps(validation_data))
+        summary = AutoApprovalSummary(version=self.version)
+        weight_info = summary.calculate_weight()
+        assert summary.weight == 40
+        assert weight_info['uses_remote_scripts'] == 40
+
+    def test_calculate_weight_violates_mozilla_conditions_of_use(self):
+        validation_data = {
+            'messages': [{
+                'id': ['MOZILLA_COND_OF_USE'],
+            }]
+        }
+        self.file_validation.update(validation=json.dumps(validation_data))
+        summary = AutoApprovalSummary(version=self.version)
+        weight_info = summary.calculate_weight()
+        assert summary.weight == 20
+        assert weight_info['violates_mozilla_conditions'] == 20
+
+    def test_calculate_weight_uses_unknown_minified_code_nothing(self):
+        validation_data = {
+            'metadata': {
+                'unknownMinifiedFiles': []  # Empty list: no weight.
+            }
+        }
+        self.file_validation.update(validation=json.dumps(validation_data))
+        summary = AutoApprovalSummary(version=self.version)
+        weight_info = summary.calculate_weight()
+        assert summary.weight == 0
+        assert weight_info['uses_unknown_minified_code'] == 0
+
+        validation_data = {
+            'metadata': {
+                # Missing property: no weight.
+            }
+        }
+        self.file_validation.update(validation=json.dumps(validation_data))
+        summary = AutoApprovalSummary(version=self.version)
+        weight_info = summary.calculate_weight()
+        assert summary.weight == 0
+        assert weight_info['uses_unknown_minified_code'] == 0
+
+        validation_data = {
+            # Missing metadata: no weight.
+        }
+        self.file_validation.update(validation=json.dumps(validation_data))
+        summary = AutoApprovalSummary(version=self.version)
+        weight_info = summary.calculate_weight()
+        assert summary.weight == 0
+        assert weight_info['uses_unknown_minified_code'] == 0
+
+    def test_calculate_weight_uses_unknown_minified_code(self):
+        validation_data = {
+            'metadata': {
+                'unknownMinifiedFiles': ['something']
+            }
+        }
+        self.file_validation.update(validation=json.dumps(validation_data))
+        summary = AutoApprovalSummary(version=self.version)
+        weight_info = summary.calculate_weight()
+        assert summary.weight == 10
+        assert weight_info['uses_unknown_minified_code'] == 10
+
```

Comment:
Ah, I wanted to ask what happens with multiple files but double checked the spreadsheet and the max is indeed 10 so that's perfectly fine.